### PR TITLE
add ComponentDao.selectByKeyWithDuplicates

### PR DESCRIPTION
**Pull request by sns-seb**
required by Dev Cockpit

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java [hunk 1]
```diff
@@ -193,6 +193,15 @@ public ComponentDto selectOrFailByKey(DbSession session, String key) {
     return Optional.fromNullable(mapper(session).selectByKey(key));
   }
 
+  /**
+   * Select all Components in DB with the specified key.
+   *
+   * Used by Developer Cockpit
+   */
+  public List<ComponentDto> selectByKeyWithDuplicates(DbSession session, String key) {
```
Comment (julienlancelot) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: In fact this query already exists : selectComponentsHavingSameKeyOrderedById (used to remove possible duplicated keys when creating a project on MySQL)

Comment (julienlancelot) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: So you could either use selectComponentsHavingSameKeyOrderedById, or replace usage of selectComponentsHavingSameKeyOrderedById by you're new one but in this case don't forget to add the order by id

Comment (sns-seb) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: I'll use the existing method, checking ITs right now on Dev Cockpit
I'll close this PR then

Comment (julienlancelot) [inline sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java]: ok

**Comment by sns-seb** [comment]
@julienlancelot I've changed the content of this PR to a maintenance commit to simplify `ComponentDaoWithDuplicatedKeysTest` by avoiding having a specific declaration of table `PROJECTS` for this test

WDYT?

**Comment by julienlancelot** [comment]
@seb ok, but do you still need selectByKeyWithDuplicates ?

**Comment by sns-seb** [comment]
@julienlancelot nop, it's deleted

**Comment by sns-seb** [comment]
but I do need a change in the existing method for developer cockpit. Currently, returned `ComponentDto` objects have only their `id` fields populated => I've added another commit to this PR

File: sonar-db/src/test/java/org/sonar/db/component/ComponentDaoWithDuplicatedKeysTest.java [hunk 2]
```diff
@@ -63,6 +73,16 @@ public void return_nothing() throws Exception {
...
   }
 
+  @Test
+  public void selectByKeyWithDuplicates_returns_all_components_with_key() {
+    underTest.insert(dbSession, new ComponentDto().setKey("key").setEnabled(false));
+    dbSession.commit();
+    underTest.insert(dbSession, new ComponentDto().setKey("key").setEnabled(true));
+    dbSession.commit();
+
+    assertThat(underTest.selectByKeyWithDuplicates(dbSession, "key")).hasSize(2);
```
Comment (julienlancelot) [inline sonar-db/src/test/java/org/sonar/db/component/ComponentDaoWithDuplicatedKeysTest.java]: selectByKeyWithDuplicates is still there, no ?

Comment (sns-seb) [inline sonar-db/src/test/java/org/sonar/db/component/ComponentDaoWithDuplicatedKeysTest.java]: indeed, my bad 

fixed

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by julienlancelot** [comment]
Ok, then LGTM !

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR Add a 'package-info.java' file to document the 'protobuf' package (org.sonarsource.sonarqube:sonar-db:target/generated-sources/protobuf/org/sonar/db/protobuf) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Pull request merged**